docs/architecture/adr-002-configv2.md

```diff
@@ -0,0 +1,109 @@
+# ADR 2: Configuration v2
+
+## Overview
+
+This ADR is very high level proposal for the overall goals and high level aspirations for configuration v2. As such the puspose is not to go overly deep into where/how different fields are defined.
+
+The examples provided in this ADR are currently not exhaustive. So do NOT pay too much attention to each field, they're meant to be more examples of the _types_ of configuration for each object.
+
+## Context
+
+The configration of k0s has organically growed from very small to something quite complex today. While that is natural we're starting to see growing number of issues stemmin from the complexity.
+
+- Lack of separation between per node configuration and cluster configuration
+- Lack of feedback in dynamic config reconciliation status
+- Lack of feedback in helm and stack applier status (should these be in the clusterconfig to begin with?)
+- No separation between used defined configuration and the configuration that is used internally
+- Lack of versioning
+- Dynamic config has several problems when it comes to reconciliation of certain fields
+- Some fields have side effects that aren't very predictable.
+- The k0s' configuration for calico doesn't really match the calico configuration
+- Config doesn't allow to disable components, it requires modifying the cmdline, specially with dynamic config, it would make sense to allow this from the configuration.
+- In dynamicConfig it's not very straightforward what can and what cannot be modified
+
+All of this results in confusions on both the uers side and also on maintainers.
+
+## Config v2 goals
+
+The maintainers have agreed on set of high level goals for config v2:
+
+### Config as Kubernetes objects
+
+Configs shall be formatted as Kubernetes object, even if we do not really store them on API. K8s object are the natural language for k0s users and it keeps the option open to actually store everything in the API.
+
+### Per node and cluster wide config separation
+
+Separate the per node configuration and cluster-wide configs into their own CRDs. This makes it clear for both the users and maintainers where to look for which config data. It also makes clear separation on which _things_ can be changed at runtime via dynamic config.
```

Comment:
> It also makes clear separation on which _things_ can be changed at runtime via dynamic config.

Does it actually make it clear which things can be changed and which cannot? Obviously changing helm chart extensions is allowed at runtime, and just as obviously changing the service CIDR range is not - but both would be part of the ClusterConfig, no?